src/quipucordsctl/commands/uninstall.py

```diff
@@ -17,7 +20,177 @@ def get_help() -> str:
+            )
+            return False
+    else:
+        logger.debug(
+            _(
+                "The uninstall command did not find"
+                " the file %(file_path)s which it expected to remove."
+            ),
+            {"file_path": file_path},
+        )
+    return True
+
+
+def remove_services() -> bool:
+    """Remove services."""
+    logger.info(
+        _("Removing the %(server_software_name)s services."),
+        {"server_software_name": settings.SERVER_SOFTWARE_NAME},
+    )
+
+    for env_file in settings.TEMPLATE_SERVER_ENV_FILENAMES:
+        env_file_path = Path(settings.SERVER_ENV_DIR) / env_file
+        if not remove_file(env_file_path):
+            return False
+
+    for unit_file in settings.TEMPLATE_SYSTEMD_UNITS_FILENAMES:
+        unit_file_path = Path(settings.SYSTEMD_UNITS_DIR) / unit_file
+        if not remove_file(unit_file_path):
+            return False
+
+    if (
+        settings.SYSTEMD_GENERATED_SERVICES_DIR
+        and settings.SYSTEMD_GENERATED_SERVICES_DIR.exists()
+    ):
+        for service_file in settings.SYSTEMD_SERVICE_FILENAMES:
+            service_file_path = settings.SYSTEMD_GENERATED_SERVICES_DIR / service_file
+            if not remove_file(service_file_path):
+                return False
+    return True
+
+
+def reload_daemon() -> bool:
+    """Reset systemctl failures and reload the daemon."""
+    logger.info(_("Reloading the systemctl daemon ..."))
+    shell_utils.run_command(settings.SYSTEMCTL_USER_RESET_FAILED_CMD)
+    shell_utils.run_command(settings.SYSTEMCTL_USER_DAEMON_RELOAD_CMD)
+    return True
+
+
+def remove_data():
+    """Remove the quipucords data."""
+    logger.info(
+        _("Removing the %(server_software_name)s data ..."),
+        {"server_software_name": settings.SERVER_SOFTWARE_NAME},
+    )
+    for data_dir in settings.SERVER_DATA_SUBDIRS_EXCLUDING_DB.values():
+        shutil.rmtree(data_dir, ignore_errors=True)
+
+
+def remove_secrets() -> bool:
+    """Remove the quipucords podman secrets."""
+    logger.info(
+        _("Removing the %(server_software_name)s secrets ..."),
+        {"server_software_name": settings.SERVER_SOFTWARE_NAME},
+    )
+    for key in settings.QUIPUCORDS_SECRET_KEYS:
+        if not podman_utils.delete_secret(key):
+            return False
+    return True
+
+
+def run(args: argparse.Namespace) -> bool:  # noqa: PLR0911
     """Uninstall the server."""
-    logger.warning(_("%(command)s is not yet implemented."), {"command": __name__})
-    # TODO Implement this.
+    if not stop_containers():
+        return False
+    remove_container_images()
+    if not remove_services():
+        return False
+    if not reload_daemon():
+        return False
+    remove_data()
+    if not remove_secrets():
+        return False
+    print(
+        _("%(server_software_name)s uninstalled successfully.")
+        % {"server_software_name": settings.SERVER_SOFTWARE_NAME}
+    )
```

Comment:
```suggestion
    if not args.quiet:
        print(
            _("%(server_software_name)s uninstalled successfully.")
            % {"server_software_name": settings.SERVER_SOFTWARE_NAME}
        )
```
Gotta respect the `quiet` unless we have a darn good reason not to. 😉 In my upcoming PR, `quiet` is even quieter, and I don't want this to slip through.